sys/include/net/dns/cache.h

```diff
@@ -34,18 +35,34 @@
 
 #include <stdint.h>
 
+#include "kernel_defines.h"
+
 #ifdef __cplusplus
 extern "C" {
 #endif
 
 /**
- * @brief   Number of DNS cache entries
+ * @brief   Maximum number of DNS cache entries
  */
 #ifndef CONFIG_DNS_CACHE_SIZE
 #define CONFIG_DNS_CACHE_SIZE   4
 #endif
 
-#if IS_USED(MODULE_SOCK_DNS_CACHE) || DOXYGEN
+/**
+ * @brief   Handle to cache A records
+ */
+#ifndef CONFIG_DNS_CACHE_A
+#define CONFIG_DNS_CACHE_A      1
```

Comment:
`SOCK_HAS_IPV4` is a feature property of `sock`. If we move this module out of `sock` this should not depend on it (as we otherwise need to include `sock` headers). I chose to only support A records as the defaults as this yields the smaller size for the overall cache size, but sure I also can default to AAAA. With my argument provided, what do you prefer?

Reply:
Will go with the modified approach proposed in https://github.com/RIOT-OS/RIOT/pull/18318#pullrequestreview-1041800304

Reply:
I'm generally not a fan of broken defaults. "To enable DNS caching, just include the `dns_cache` module. Oh but you also have to set CFLAGS otherwise it will silently not work" :wink: